src/test/ui/lint/pineapple-on-pizza.rs

```diff
@@ -0,0 +1,6 @@
+struct Pizza<T>(T);
+struct Pineapple;
+
+fn main() {
+    let _: Pizza<Pineapple>; //~ERROR pineapple doesn't go on pizza
```

Comment:
A test is necessary to ensure that this lint fires regardless of the order in which – or count of – ingredients are applied to the base.